packages/plugin-exec/sources/ExecFetcher.ts

```diff
@@ -1,9 +1,40 @@
-import {Fetcher, FetchOptions, MinimalFetchOptions}                    from '@yarnpkg/core';
-import {Locator, MessageName}                                          from '@yarnpkg/core';
-import {execUtils, scriptUtils, structUtils, tgzUtils}                 from '@yarnpkg/core';
-import {Filename, NodeFS, PortablePath, npath, ppath, toFilename, xfs} from '@yarnpkg/fslib';
-
-import {PROTOCOL}                                                      from './constants';
+import {execUtils, scriptUtils, structUtils, tgzUtils}                     from '@yarnpkg/core';
+import {Locator, MessageName}                                              from '@yarnpkg/core';
+import {Fetcher, FetchOptions, MinimalFetchOptions}                        from '@yarnpkg/core';
+import {Filename, PortablePath, npath, ppath, toFilename, xfs, NativePath} from '@yarnpkg/fslib';
+
+import {PROTOCOL}                                                          from './constants';
+import {getGeneratorPath}                                                  from './execUtils';
+
+/**
+ * Contains various useful details about the execution context.
+ */
+export interface ExecEnv {
+  /**
+   * The absolute path of the temporary directory where the script runs. Equal to `process.cwd()`.
+   */
+  tempDir: NativePath;
```

Comment:
I think it should point to a subdirectory of the "global" temporary directory, so that there is no risk for build to conflict with something else (for example when cloning a repository with an actual `build` subdirectory).